packages/identity/src/idp-identity.ts

```diff
@@ -0,0 +1,339 @@
+   */
+  readonly cognitoDomain?: CognitoDomainOptions;
+
+  /**
+   * Options while specifying custom domain
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-add-custom-domain.html
+   */
+  readonly customDomain?: CustomDomainOptions;
+
+  /**
+   * Configuration for the Identity Pool.
+   */
+  readonly identityPoolOptions?: IdentityPoolProps;
+
+  /**
+   * Configuration for the Federated Identity Providers.
+   */
+  readonly identityProviderProps?: IdentityProviderProps;
+
+  /**
+   * Options to create a UserPoolClient
+   */
+  readonly userPoolClientProps?: UserPoolClientOptions;
+
+  readonly identityProviders?: IdentityProviderProps;
+}
+
+/**
+ * Properties which configure a user pool client.
+ */
+export interface AddClientProps extends UserPoolClientOptions {
+  /**
+   * List of allowed redirect URLs for the identity providers.
+   * @default - ['https://example.com'] if either authorizationCodeGrant or implicitCodeGrant flows are enabled, no callback URLs otherwise.
+   */
+  readonly callbackUrls: Array<string>;
+
+  /**
+   * List of allowed logout URLs for the identity providers.
+   * @default - no logout URLs
+   */
+  readonly logoutUrls: Array<string>;
+
+  /**
+   * Options while specifying a cognito prefix domain.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-assign-domain-prefix.html
+   */
+  readonly cognitoDomain?: CognitoDomainOptions;
+
+  /**
+   * Options while specifying custom domain
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-add-custom-domain.html
+   */
+  readonly customDomain?: CustomDomainOptions;
+
+  readonly useIdentityProvider?: Array<UserPoolClientIdentityProvider>;
+}
+
+/**
+ * Properties which configure a domain for the user pool.
+ */
+interface CreateUserPoolProps extends UserPoolProps {
+  readonly cognitoDomain?: CognitoDomainOptions;
+  readonly customDomain?: CustomDomainOptions;
+}
+
+/**
+ * Creates a user pool will one or more federated identity providers
+ */
+export class IdpIdentity extends Construct {
+  public userPool: IUserPool;
+  public userPoolClient?: UserPoolClient;
+  protected hasIdentityPool: boolean = false;
+  protected identityProviders: Array<IUserPoolIdentityProvider> = [];
+
+  constructor(scope: Construct, id: string, readonly props?: IdpIdentityProps) {
+    super(scope, id);
+    Annotations.of(this).addInfo(`${JSON.stringify(props)}`);
+
+    if (props?.userPool) {
+      this.userPool = props.userPool!;
+    } else {
+      this.userPool = this._createUserPool(id, props);
+      this._addDomain(id, this.userPool, {
+        cognitoDomain: props?.cognitoDomain,
+        customDomain: props?.customDomain,
+      });
+      this._attachIdentityProviders(
+        id,
+        props?.identityProviders!,
+        this.userPool
+      );
+    }
+  }
```

Comment:
Is it expected behaviour that the domain and identity providers don't get set up if you pass in your own user pool? I think this needs more detailed documentation if so :)